Library/Homebrew/github_packages.rb

```diff
@@ -235,10 +235,18 @@ def upload_bottle(user, token, skopeo, formula_full_name, bottle_hash, dry_run:)
       end
       raise TypeError, "unknown tab['built_on']['os']: #{tab["built_on"]["os"]}" if os.blank?
 
-      os_version = if tab["built_on"].present? && tab["built_on"]["os_version"].present?
-        tab["built_on"]["os_version"]
+      os_version = tab["built_on"]["os_version"] if tab["built_on"].present?
+      os_version = case os
+      when "darwin"
+        os_version || "macOS #{MacOS::Version.from_symbol(bottle_tag)}"
+      when "linux"
+        glibc_version = tab["built_on"]["glibc_version"] if tab["built_on"].present?
+        os_version ||= "Ubuntu 16.04.7"
+        glibc_version ||= "2.23"
+        os_version = os_version.delete_suffix " LTS"
+        "#{os_version} glibc #{glibc_version}"
```

Comment:
This isn't part of the OS version. I'd support it being added as additional metadata.

Reply:
I've moved it to the annotation `sh.brew.bottle.glibc.version`.